code_producers/src/llvm_elements/functions.rs

```diff
@@ -77,16 +77,16 @@ impl<'a> FunctionCtx<'a> {
 }
 
 impl<'a> BodyCtx<'a> for FunctionCtx<'a> {
-    fn get_variable(
+    fn get_lvar_ref(
         &self,
         producer: &dyn LLVMIRProducer<'a>,
         index: IntValue<'a>,
-    ) -> AnyValueEnum<'a> {
+    ) -> PointerValue<'a> {
         create_gep(producer, self.arena, &[index])
     }
 
-    fn get_variable_array(&self, _producer: &dyn LLVMIRProducer<'a>) -> AnyValueEnum<'a> {
-        self.arena.into()
+    fn get_variable_array(&self, _producer: &dyn LLVMIRProducer<'a>) -> PointerValue<'a> {
+        self.arena
```

Comment:
This is unrelated to this PR, but could we also output the arena size as part of the JSON summary?

Reply:
That one's a bit tricky because the arena size is not computed yet but I'll look into it.